python/shotgun_model/shotgun_model.py

```diff
@@ -267,6 +269,16 @@ def get_entity_type(self):
         """
         return self.__entity_type
 
+    def get_additional_column_fields(self):
+        """
+        Returns the fields for additional columns and their associated column in the model.
+
+        :returns: A list of (column, field) tuples where column is the column number in the
+            model associated with the additional field.
+        """
+        # column is one greater than the index because of the default initial column
+        return [(i+1, field) for (i, field) in enumerate(self.__column_fields)]
```

Comment:
A richer interface makes sense.  I didn't add more info to the return value but changing to a dict with keys for column_idx and field.